lib/ts/recipe/webauthn/api/implementation.ts

```diff
@@ -0,0 +1,1095 @@
+                AuthUtils.getErrorStatusResponseWithReason(postAuthChecks, errorCodeMap, "SIGN_UP_NOT_ALLOWED");
+                throw new Error("This should never happen");
+            }
+
+            return {
+                status: "OK",
+                session: postAuthChecks.session,
+                user: postAuthChecks.user,
+            };
+        },
+
+        signInPOST: async function ({
+            webauthnGeneratedOptionsId,
+            credential,
+            tenantId,
+            session,
+            shouldTryLinkingWithSessionUser,
+            options,
+            userContext,
+        }: {
+            webauthnGeneratedOptionsId: string;
+            credential: CredentialPayload;
+            tenantId: string;
+            session?: SessionContainerInterface;
+            shouldTryLinkingWithSessionUser: boolean | undefined;
+            options: APIOptions;
+            userContext: UserContext;
+        }): Promise<
+            | {
+                  status: "OK";
+                  session: SessionContainerInterface;
+                  user: User;
+              }
+            | { status: "WRONG_CREDENTIALS_ERROR" }
+            | {
+                  status: "SIGN_IN_NOT_ALLOWED";
+                  reason: string;
+              }
+            | GeneralErrorResponse
+        > {
+            const errorCodeMap = {
+                SIGN_IN_NOT_ALLOWED:
+                    "Cannot sign in due to security reasons. Please try recovering your account, use a different login method or contact support. (ERR_CODE_008)",
+                LINKING_TO_SESSION_USER_FAILED: {
+                    EMAIL_VERIFICATION_REQUIRED:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_009)",
+                    RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_010)",
+                    ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_011)",
+                    SESSION_USER_ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_012)",
+                },
+            };
+
+            const recipeId = "webauthn";
+
+            // do the verification before in order to retrieve the user email
+            const verifyCredentialsResponse = await options.recipeImplementation.verifyCredentials({
+                credential,
+                webauthnGeneratedOptionsId,
+                tenantId,
+                userContext,
+            });
+            const checkCredentialsOnTenant = async () => {
+                return verifyCredentialsResponse.status === "OK";
+            };
```

Comment:
Is this correct? is credential validation independent of the tenant?

Reply:
Done. 

Was doing it like this, because at the moment of implementation, we hadn't decided on having a method for retrieving the `webauthGeneratedOptions` (which contained the email) and i had to retrieve the email before calling `AuthUtils.getAuthenticatingUserAndAddToCurrentTenantIfRequired`. 
In the last few chats with Rishabh we deciede on adding the new method, but didn't update the API implmentation yet.

As a side note:
The credential can be shared across tenants and is not linked to a tenant.

Reply:
I see. So the point of `checkCredentialsOnTenant` is to find out if the user can use the current set of credentials to log in on the given `tenantId` (it can take it as a first parameter). This is tenant independent for pwless and thirdparty, but not for emailpassword for example.

Reply:
reverted to previous version because we don't need to check in the scope of the tenant.

Reply:
done